--- a/Makefile.am
+++ b/Makefile.am
@@ -204,6 +204,14 @@ if ENABLE_ORACLE
 SUBDIRS += plugins/omoracle
 endif
 
+if ENABLE_IMZEROMQ
+SUBDIRS += plugins/imzeromq
+endif
+
+if ENABLE_OMZEROMQ
+SUBDIRS += plugins/omzeromq
+endif
+
 if ENABLE_GUI
 SUBDIRS += java
 endif
@@ -236,6 +244,8 @@ DISTCHECK_CONFIGURE_FLAGS=	--enable-gssapi_krb5 \
 				--enable-pmlastmsg \
 				--enable-omruleset \
 				--enable-omprog \
+				--enable-imzeromq \
+				--enable-omzeromq \
 				--enable-imdiag \
 				--enable-imptcp \
 				--enable-imttcp \
--- a/configure.ac
+++ b/configure.ac
@@ -1100,7 +1100,30 @@ AC_ARG_ENABLE(omdbalerting,
 )
 AM_CONDITIONAL(ENABLE_OMDBALERTING, test x$enable_omdbalerting = xyes)
 
+# settings for the ZeroMQ input module
+AC_ARG_ENABLE(imzeromq,
+        [AS_HELP_STRING([--enable-imzeromq],[file input module enabled @<:@default=no@:>@])],
+        [case "${enableval}" in
+         yes) enable_imzeromq="yes" ;;
+          no) enable_imzeromq="no" ;;
+           *) AC_MSG_ERROR(bad value ${enableval} for --enable-imzeromq) ;;
+         esac],
+        [enable_imzeromq=no]
+)
+AM_CONDITIONAL(ENABLE_IMZEROMQ, test x$enable_imzeromq = xyes)
+
 
+# settings for the omzeromq output module
+AC_ARG_ENABLE(omzeromq,
+        [AS_HELP_STRING([--enable-omzeromq],[Compiles omzeromq module @<:@default=no@:>@])],
+        [case "${enableval}" in
+         yes) enable_omzeromq="yes" ;;
+          no) enable_omzeromq="no" ;;
+           *) AC_MSG_ERROR(bad value ${enableval} for --enable-omzeromq) ;;
+         esac],
+        [enable_omzeromq=no]
+)
+AM_CONDITIONAL(ENABLE_OMZEROMQ, test x$enable_omzeromq = xyes)
 
 # building the GUI (mostly for diagnostic reasons)
 AC_ARG_ENABLE(gui,
@@ -1259,6 +1282,8 @@ AC_CONFIG_FILES([Makefile \
 		plugins/sm_cust_bindcdr/Makefile \
 		plugins/mmsnmptrapd/Makefile \
 		plugins/cust1/Makefile \
+		plugins/omzeromq/Makefile \
+		plugins/imzeromq/Makefile \
 		java/Makefile \
 		tests/Makefile])
 AC_OUTPUT
@@ -1284,6 +1309,7 @@ echo "    imdiag enabled:                           $enable_imdiag"
 echo "    file input module enabled:                $enable_imfile"
 echo "    Solaris input module enabled:             $enable_imsolaris"
 echo "    periodic statistics module enabled:       $enable_impstats"
+echo "    ZeroMQ input module enabled:              $enable_imzeromq"
 echo
 echo "---{ output plugins }---"
 echo "    Mail support enabled:                     $enable_mail"
@@ -1296,6 +1322,7 @@ echo "    omruleset module will be compiled:        $enable_omruleset"
 echo "    omdbalerting module will be compiled:     $enable_omdbalerting"
 echo "    omudpspoof module will be compiled:       $enable_omudpspoof"
 echo "    omuxsock module will be compiled:         $enable_omuxsock"
+echo "    omzeromq module will be compiled:         $enable_omzeromq"
 echo
 echo "---{ parser modules }---"
 echo "    pmrfc3164sd module will be compiled:      $enable_pmrfc3164sd"
