--- a/Makefile.am
+++ b/Makefile.am
@@ -188,6 +188,14 @@ if ENABLE_OMPROG
 SUBDIRS += plugins/omprog
 endif
 
+if ENABLE_IMZEROMQ
+SUBDIRS += plugins/imzeromq
+endif
+
+if ENABLE_OMZEROMQ
+SUBDIRS += plugins/omzeromq
+endif
+
 if ENABLE_RFC3195
 SUBDIRS += plugins/im3195
 endif
@@ -227,6 +235,8 @@ DISTCHECK_CONFIGURE_FLAGS=	--enable-gssapi_krb5 \
 				--enable-pmlastmsg \
 				--enable-omruleset \
 				--enable-omprog \
+				--enable-imzeromq \
+				--enable-omzeromq \
 				--enable-imdiag \
 				--enable-imptcp \
 				--enable-omuxsock \
--- a/configure.ac
+++ b/configure.ac
@@ -846,6 +846,19 @@ AC_ARG_ENABLE(imfile,
 AM_CONDITIONAL(ENABLE_IMFILE, test x$enable_imfile = xyes)
 
 
+# settings for the ZeroMQ input module
+AC_ARG_ENABLE(imzeromq,
+        [AS_HELP_STRING([--enable-imzeromq],[file input module enabled @<:@default=no@:>@])],
+        [case "${enableval}" in
+         yes) enable_imzeromq="yes" ;;
+          no) enable_imzeromq="no" ;;
+           *) AC_MSG_ERROR(bad value ${enableval} for --enable-imzeromq) ;;
+         esac],
+        [enable_imzeromq=no]
+)
+AM_CONDITIONAL(ENABLE_IMZEROMQ, test x$enable_imzeromq = xyes)
+
+
 # settings for the door input module (under solaris, thus default off)
 AC_ARG_ENABLE(imsolaris,
         [AS_HELP_STRING([--enable-imsolaris],[solaris input module enabled @<:@default=no@:>@])],
@@ -897,6 +910,19 @@ AC_ARG_ENABLE(omprog,
 AM_CONDITIONAL(ENABLE_OMPROG, test x$enable_omprog = xyes)
 
 
+# settings for the omzeromq output module
+AC_ARG_ENABLE(omzeromq,
+        [AS_HELP_STRING([--enable-omzeromq],[Compiles omzeromq module @<:@default=no@:>@])],
+        [case "${enableval}" in
+         yes) enable_omzeromq="yes" ;;
+          no) enable_omzeromq="no" ;;
+           *) AC_MSG_ERROR(bad value ${enableval} for --enable-omzeromq) ;;
+         esac],
+        [enable_omzeromq=no]
+)
+AM_CONDITIONAL(ENABLE_OMZEROMQ, test x$enable_omzeromq = xyes)
+
+
 # settings for omudpspoof
 AC_ARG_ENABLE(omudpspoof,
         [AS_HELP_STRING([--enable-omudpspoof],[Compiles omudpspoof module @<:@default=no@:>@])],
@@ -1167,6 +1193,7 @@ AC_CONFIG_FILES([Makefile \
 		plugins/omtemplate/Makefile \
 		plugins/omhdfs/Makefile \
 		plugins/omprog/Makefile \
+		plugins/omzeromq/Makefile \
 		plugins/omstdout/Makefile \
 		plugins/pmrfc3164sd/Makefile \
 		plugins/pmlastmsg/Makefile \
@@ -1177,6 +1204,7 @@ AC_CONFIG_FILES([Makefile \
 		plugins/omdbalerting/Makefile \
 		plugins/omuxsock/Makefile \
 		plugins/imfile/Makefile \
+		plugins/imzeromq/Makefile \
 		plugins/imsolaris/Makefile \
 		plugins/imptcp/Makefile \
 		plugins/impstats/Makefile \
@@ -1216,6 +1244,7 @@ echo "    Klog functionality enabled:               $enable_klog ($os_type)"
 echo "    plain tcp input module enabled:           $enable_imptcp"
 echo "    imdiag enabled:                           $enable_imdiag"
 echo "    file input module enabled:                $enable_imfile"
+echo "    ZeroMQ input module enabled:              $enable_imzeromq"
 echo "    Solaris input module enabled:             $enable_imsolaris"
 echo "    periodic statistics module enabled:       $enable_impstats"
 echo "    input template module will be compiled:   $enable_imtemplate"
@@ -1223,6 +1252,7 @@ echo
 echo "---{ output plugins }---"
 echo "    Mail support enabled:                     $enable_mail"
 echo "    omprog module will be compiled:           $enable_omprog"
+echo "    omzeromq module will be compiled:         $enable_omzeromq"
 echo "    omstdout module will be compiled:         $enable_omstdout"
 echo "    omhdfs module will be compiled:           $enable_omhdfs"
 echo "    omruleset module will be compiled:        $enable_omruleset"
